third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -1742,19 +1745,31 @@ impl Loader {
                 .into_iter()
                 .unzip();
 
+        // Some types can have fields which have custom layouts at serialization or
+        // deserialization times. If so, they must be tagged.
+        let maybe_layout_tag = self.native_types.get_layout_tag(name);
         let has_identifier_mappings =
-            maybe_mapping.is_some() || field_has_identifier_mappings.into_iter().any(|b| b);
+            maybe_layout_tag.is_some() || field_has_identifier_mappings.into_iter().any(|b| b);
 
         let field_node_count = *count - count_before;
-        let layout = if Some(IdentifierMappingKind::DerivedString) == maybe_mapping {
+
+        // Construct struct layout.
+        // TODO[agg_v2](cleanup): Once we have a native value impl, we can also
+        //   construct "native", i.e. tagged layouts for these native fields, such
+        //   as MoveTypeLayout::Native(u64) which are fast to propagate. This can
```

Comment:
Didn't understand the comment. What does MoveTypeLayout::Native(u64) mean? How does it differ from MoveTypeLayout::u64?

Reply:
What I was trying to say is that we can register a layout directly, and MoveVM will see a `MoveTypeLayout::Native(v: u64)` where `v` is a handle which can be used to fetch the layout directly. It is a bit difficult to achieve because of generics though